### PR TITLE
CDGPS Sensor Model

### PR DESCRIPTION
**Pull request by kylekrol**
Relates to #141.

### Summary of changes
- Simple range based CDGPS sensor with additive gaussian noise.
- Sensor plotting files.
- Updated default PSim configurations.
- Added CDGPS sensors to all dual cases.

### Testing

The CDGPS sensor model was tested using deployment initial conditions to verify the range based sensor behavior:

    python -m psim -s 60000 -p sensors/cdgps,sensors/gps -ps 10 -c sensors/base,truth/base,truth/deployment -v DualOrbitGnc

Notice how a little over 8000 seconds in the CDGPS sensor readings disappear where as GPS readings don't; this means the sensor was deactivate at longer ranges as expected.

cdgps
cdgps_error
gps

### CODE REVIEW

**Comment by kylekrol** [comment]
@shihaocao and whoever else is familiar with RTK, if we could verify exactly what they mean by baseline position for this PR that would be much appreciated!

**Comment by kylekrol** [comment]
@shihaocao Per our brief discussion of RTK readouts, once I flip a minus sign I believe this model should be correct. We take the different between the two satellites in ECEF and return the vector pointing from the "other" satellite to "this" satellite.

File: src/psim/sensors/cdgps_no_attitude.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Vector3 CdgpsNoAttitude::sensors_satellite_cdgps_dr() const {
+  auto const &truth_r_ecef = truth_satellite_orbit_r_ecef->get();
+  auto const &truth_other_r_ecef = truth_other_orbit_r_ecef->get();
+  auto const &error = Super::sensors_satellite_cdgps_dr_error.get();
+
+  /** If the sensor isn't active, the NaNs from the error field will make this
+   *  calculation results in NaNs as well as desired.
+   */
+  return (truth_other_r_ecef - truth_r_ecef) + error;
```
Comment (kylekrol) [inline src/psim/sensors/cdgps_no_attitude.cpp]:
```suggestion
  return (truth_r_ecef - truth_other_r_ecef) + error;
```

File: include/psim/sensors/cdgps_no_attitude.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+adds:
+    - name: "sensors.{satellite}.cdgps.active"
+      type: Lazy Integer
+      comment: >
+          Flag specifying whether or not the CDGPS sensor is currently active
+          and producing readings.
+    - name: "sensors.{satellite}.cdgps.dr"
+      type: Lazy Vector3
+      comment: >
+          Relative position of the other satellite reported by the CDGPS in ECEF.
```
Comment (kylekrol) [inline include/psim/sensors/cdgps_no_attitude.yml]:
```suggestion
          Relative position this satellite to the other reported by the CDGPS in
          ECEF.
```

File: include/psim/sensors/cdgps_no_attitude.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - name: "sensors.{satellite}.cdgps.dr.error"
+      type: Lazy Vector3
+      comment: >
+          Error in the relative position of the other satellite by the CDGPS in
+          ECEF.
```
Comment (kylekrol) [inline include/psim/sensors/cdgps_no_attitude.yml]:
```suggestion
          Error in the relative position of the this satellite to the other
          satellite by the CDGPS in ECEF.
```

**Review by shihaocao — APPROVED** [review]

**Pull request merged**